lib/firewall/firewall.cc

```diff
@@ -12,6 +12,19 @@
 #include <timeout.hh>
 #include <vector>
 
+namespace
+{
+	// TODO These two should probably be in their own library.
+	uint16_t ntohs(uint16_t value)
+	{
+		return __builtin_bswap16(value);
```

Comment:
We currently support only little endian (and probably will never support big endian), but these should probably be wrapped in a check for `__LITTLE_ENDIAN__`.  They should probably also be `constexpr`.